core/src/Data/Repository/RepositoryMapper.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Harmony\Core\Data\Repository;
+
+use Harmony\Core\Data\Mapper\Mapper;
+use Harmony\Core\Data\Operation\Operation;
+use Harmony\Core\Data\Query\Query;
+
+/**
+ * @template   T
+ * @template   TModel
+ * @template   TEntity
+ * @implements GetRepository<TModel>
+ * @implements PutRepository<TModel>
+ */
+class RepositoryMapper implements
+  GetRepository,
+  PutRepository,
+  DeleteRepository {
+  /**
+   * @param GetRepository<TEntity>  $getRepository
+   * @param PutRepository<TEntity>  $putRepository
+   * @param DeleteRepository        $deleteRepository
+   * @param Mapper<TModel, TEntity> $modelToEntityMapper
+   * @param Mapper<TEntity, TModel> $entityToModelMapper
+   */
+  public function __construct(
+    protected readonly GetRepository $getRepository,
+    protected readonly PutRepository $putRepository,
+    protected readonly DeleteRepository $deleteRepository,
+    protected readonly Mapper $modelToEntityMapper,
+    protected readonly Mapper $entityToModelMapper,
+  ) {
+  }
+
+  /**
+   * @return T
+   */
+  public function get(Query $query, Operation $operation): mixed {
+    $toMap = $this->getRepository->get($query, $operation);
+
+    if (!is_array($toMap)) {
```

Comment:
We should not make this distinction here. This is the problem of the mapper, we should be able to send the result to the mapper and the mapper will be an ArrayMapper or a single-entity mapper depending on the DI (in the provider)

Reply:
Fixed.